### PR TITLE
feat(protocol-designer): add support for TC cycles

### PR DESCRIPTION
**Pull request by IanLondon**
## overview

Allow adding/deleting/editing cycles, and adding/deleting/editing steps that are within cycles!

Pairing credit to @shlokamin 

Addresses the behavioral parts of #5516 but the styling will require another PR, same for #5517 and #5518

## review requests

- [ ] Adding a step directly (to the top level) should work as before
- [ ] Top level steps should still do validation + masking on temperature field (duration field validation not yet implemented)
- [ ] Adding a cycle should work (but look super ugly)
- [ ] Adding a step to a cycle should work (also ugly looking)
- [ ] Top level steps should still do validation + masking on temperature field (same as top-level steps, including disabling the form's save button)
- [ ] Deleting a step that is in a cycle should work
- [ ] Deleting a cycle should work (and delete all its inner steps)

- Code + tests review

## risk assessment

low, PD only under FF

### CODE REVIEW

File: protocol-designer/src/form-types.js [hunk 1]
```diff
@@ -159,6 +159,7 @@ export type ProfileCycleItem = {|
   repetitions: string,
 |}
 
+// TODO IMMEDIATELY: ProfileStepItem -> ProfileStep, ProfileCycleItem -> ProfileCycle
```
Comment (IanLondon) [inline protocol-designer/src/form-types.js:162]: Follow-up PR. Shlok & I figured this renaming would make these types less confusing

File: protocol-designer/src/step-forms/reducers/index.js [hunk 2]
```diff
@@ -97,9 +103,28 @@ import type {
 
 type FormState = FormData | null
 
+// TODO IMMEDIATELY factor out
```
Comment (IanLondon) [inline protocol-designer/src/step-forms/reducers/index.js:106]: woops missed these, will do tomorrow

**Review by shlokamin — APPROVED** [review]
🌀  🌀  🌀

**Review by Kadee80 — APPROVED** [review]
🧄

**Pull request merged**